airflow/models/dag.py

```diff
@@ -2296,84 +2293,8 @@ def _remove_task(self, task_id: str) -> None:
-        ignore_task_deps=False,
-        ignore_first_depends_on_past=True,
-        pool=None,
-        delay_on_limit_secs=1.0,
-        verbose=False,
-        conf=None,
-        rerun_failed_tasks=False,
-        run_backwards=False,
-        run_at_least_once=False,
-        continue_on_failures=False,
-        disable_retry=False,
-    ):
-        """
-        Run the DAG.
-
-        :param start_date: the start date of the range to run
-        :param end_date: the end date of the range to run
-        :param mark_success: True to mark jobs as succeeded without running them
-        :param local: True to run the tasks using the LocalExecutor
-        :param donot_pickle: True to avoid pickling DAG object and send to workers
-        :param ignore_task_deps: True to skip upstream tasks
-        :param ignore_first_depends_on_past: True to ignore depends_on_past
-            dependencies for the first set of tasks only
-        :param pool: Resource pool to use
-        :param delay_on_limit_secs: Time in seconds to wait before next attempt to run
-            dag run when max_active_runs limit has been reached
-        :param verbose: Make logging output more verbose
-        :param conf: user defined dictionary passed from CLI
-        :param rerun_failed_tasks:
-        :param run_backwards:
-        :param run_at_least_once: If true, always run the DAG at least once even
-            if no logical run exists within the time range.
-        """
-        warnings.warn(
-            "`DAG.run()` is deprecated and will be removed in Airflow 3.0. Consider "
-            "using `DAG.test()` instead, or trigger your dag via API.",
-            RemovedInAirflow3Warning,
-            stacklevel=2,
-        )
-
-        from airflow.executors.executor_loader import ExecutorLoader
-        from airflow.jobs.backfill_job_runner import BackfillJobRunner
-
-        if local:
-            from airflow.executors.local_executor import LocalExecutor
-
-            ExecutorLoader.set_default_executor(LocalExecutor())
-
-        from airflow.jobs.job import Job
-
-        job = Job()
-        job_runner = BackfillJobRunner(
-            job=job,
-            dag=self,
-            start_date=start_date,
-            end_date=end_date,
-            mark_success=mark_success,
-            donot_pickle=donot_pickle,
-            ignore_task_deps=ignore_task_deps,
-            ignore_first_depends_on_past=ignore_first_depends_on_past,
-            pool=pool,
-            delay_on_limit_secs=delay_on_limit_secs,
-            verbose=verbose,
-            conf=conf,
-            rerun_failed_tasks=rerun_failed_tasks,
-            run_backwards=run_backwards,
-            run_at_least_once=run_at_least_once,
-            continue_on_failures=continue_on_failures,
-            disable_retry=disable_retry,
-        )
-        run_job(job=job, execute_callable=job_runner._execute)
+    def run(self, *args, **kwargs):
```

Comment:
Isn't it easier to just remove this too?

Reply:
there are things that mock it as part of testing of backfill. but that is easier to remove when we will remove / replace legacy backfill.  but doing that in this PR together is just too much IMO